### ISSUE TITLE
Event stream table size limit?

### ISSUE THREAD

**Issue by suzdalnitski** [commanded/eventstore#256]
Given enough rows in a relational database table, the performance is likely going to start degrading. How does eventstore handle this internally? Does it make use of table partitioning?

What is the scale that eventstore can handle?

For example, if an application appends 100 events per second, this is going to result in 260,000,000 rows in a month (and about 3 billion in a year). Is this something that eventstore is going to be able to handle?

**Comment by suzdalnitski** [comment]
I did an experiment to answer my own question - I've been appending events for a few days, and measuring the time it took to append an event. I got to about 600m events before I started running out of disk space, and here are the results (for the append operation):

![image](https://user-images.githubusercontent.com/8169977/205510810-1df4c4c1-0848-4a62-826e-599b290a6e16.png)

And the mean:
![image](https://user-images.githubusercontent.com/8169977/205510872-538c8797-a7c3-483b-b864-5591a41e8899.png)

It seems that the append operation becomes fairly linear after about 100m events:
- At 1m total events it takes 72 usecs to append one event
- At 10m it takes 90 usecs
- At 100m it takes 136 usecs
- At 200m it takes 220 usecs
- At 300m it takes 329 usecs
- At 400m it takes 398 usecs
- At 500m it takes 493 usecs
- At 600m it takes 550 usecs

Extrapolating forward, it might take about *1ms to append one event at 1bln* total events, and *10ms at 10bln events*.

In my own experiments with Commanded, dispatching an event takes about 15ms (strong consistency, including rebuilding the read models), so 1ms @ 1bln is negligible, however the difference may become noticeable once we get to about 10bln events. Of course, Postgres can handle much more, especially with partitioning/AuroraDB.

@slashdotdash I guess the tables can't be easily partitioned, so sharding probably is the preferred approach with a large enough number of events?
